static/js/setup.js

```diff
@@ -69,12 +70,18 @@ $(() => {
         }
 
         if (xhr.status === 401) {
-            // We got logged out somehow, perhaps from another window
-            // changing the user's password, or a session timeout.  We
-            // could display an error message, but jumping right to
-            // the login page conveys the same information with a
-            // smoother re-login experience.
-            window.location.replace(page_params.login_page);
+            if (page_params.is_web_public_visitor) {
+                // We should aim to never reach here, since all the login_to_access
+                // cases should be handled in hashchange.js before requesting server.
+                login_to_access.show();
```

Comment:
Can you talk about example situations where this will happen?

Reply:
Right now, I am handing all the requests that a web public visitor shouldn't have access to before it reaches the server to show login_to_access modal. But, we can introduce a new something in future which tries to fetch data which isn't allowed for web public visitor users. This is to handle for that case.